opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -235,6 +235,19 @@ namespace Opm {
             well->calculateReservoirRates(well_state_);
         }
         updateWellTestState(simulationTime, wellTestState_);
+
+        // calculate the well potentials for output
+        // TODO: when necessary
+        try
+        {
```

Comment:
I know it is good to calculate based on the summary output requirement now.  Maybe before that, you can keep the original one `if (wellCollection().requireWellPotentials())` to avoid to calculate well potentials for all cases? It is not something very cheap, although not huge either.  And also, the calculating well potentials based on THP is still buggy, it might cause crashes for those cases.